### ISSUE TITLE
Integer overflow in io_loop.rs

### ISSUE THREAD

**Issue by dylanhart** [sozu-proxy/lapin#265]
I encountered the following when attempting to connect to an amqp server with the default settings.

```rust
let conn = Connection::connect(&server_uri, ConnectionProperties::default()).await?;
```

```
thread 'io_loop' panicked at 'attempt to multiply with overflow', /home/dylanh/.cargo/registry/src/github.com-1ecc6299db9ec823/lapin-1.0.0-alpha1/src/io_loop.rs:113:65
```

I was able to patch io_loop in order to work around the issue.

### Suggested Patch
```diff
diff --git a/src/io_loop.rs b/src/io_loop.rs
index 53044b2..a2b8581 100644
--- a/src/io_loop.rs
+++ b/src/io_loop.rs
@@ -110,7 +110,7 @@ impl<T: Source + Read + Write + Send + 'static> IoLoop<T> {
             self.send_buffer.grow(FRAMES_STORAGE * self.frame_size);
             let heartbeat = self.configuration.heartbeat();
             if heartbeat != 0 {
-                let heartbeat = Duration::from_millis(u64::from(heartbeat * 500)); // * 1000 (ms) / 2 (half the negociated timeout)
+                let heartbeat = Duration::from_millis(u64::from(heartbeat) * 500); // * 1000 (ms) / 2 (half the negociated timeout)
                 self.poll_timeout = Some(heartbeat);
             }
             self.status = Status::Setup;
```

**Comment by Keruspe** [comment]
Huh, oh, feels right indeed. Your rabbitmq server must have a quite high heartbeat I guess, thanks for hitting this

**Comment by Keruspe** [comment]
Published 0.39.1
